### PR TITLE
Add MGTwoLevelTransfer::fine_element_is_continuous and add missing compression step

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: tests/multigrid-global-coarsening/mg_transfer_a_01.with_p4est=true.mpirun=2.output [hunk 1]
```diff
@@ -5,13 +5,13 @@ DEAL:0:CG<2>(1)<->CG<2>(1)::1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 1.
 DEAL:0:CG<2>(1)<->CG<2>(1)::5.6789083
 DEAL:0:CG<2>(1)<->CG<2>(1)::1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 1.0000000 
 DEAL:0:CG<2>(1)<->CG<2>(1)::2.2500000 1.7500000 3.0000000 2.0000000 1.0000000 1.0000000 2.2500000 1.7500000 1.0000000 
-DEAL:0:DG<2>(1)<->CG<2>(1)::4.4721360
+DEAL:0:DG<2>(1)<->CG<2>(1)::6.3245553
```
Comment (peterrum) [inline tests/multigrid-global-coarsening/mg_transfer_a_01.with_p4est=true.mpirun=2.output:8]: The value is now consistent with the serial case.

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 2]
```diff
@@ -288,6 +281,13 @@ class MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
    */
   std::vector<MGTransferScheme> schemes;
 
+  /**
+   * Flag if the finite element on the fine cells are continuous. If yes,
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:285]:
```suggestion
   * Flag if the finite element on the fine cells is continuous. If yes,
```

File: include/deal.II/multigrid/mg_transfer_global_coarsening.h [hunk 3]
```diff
@@ -288,6 +281,13 @@ class MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>
...
+   * the multiplicity of DoF sharing a vertex/line as well as constraints have
+   * to be taken in account via weights.
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.h:287]:
```suggestion
   * to be taken into account via weights.
```

**Review by kronbichler — APPROVED** [review]

**Comment by peterrum** [comment]
Done!

**Review by kronbichler — APPROVED** [review]
Thank you!

**Pull request merged**